pkg/kubelet/kubelet_test.go

```diff
@@ -1436,18 +1439,31 @@ func TestFilterOutTerminatedPods(t *testing.T) {
 			},
 		}},
 	}
+
+	// pending and running pods are included
 	pods[3].Status.Phase = v1.PodPending
 	pods[4].Status.Phase = v1.PodRunning
 
-	kubelet.podWorkers.(*fakePodWorkers).running = map[types.UID]bool{
-		pods[2].UID: true,
-		pods[3].UID: true,
-		pods[4].UID: true,
+	// pod that is running but has been rejected by admission is excluded
```

Comment:
It can happen on a kubelet restart.  There's nothing "special" about hard admission except that the kubelet never reruns it outside of a restart, but must run it on a restart, and there's no real guarantee that something allowed preivously is allowed on restart.  Concrete example - the restart is to change config flags on the kubelet that then cause the pod node selector labels to be wrong.

Reply:
I see, thanks for explanation, makes sense.